projects/noloan/common/src/main/java/hasadna/noloan/common/DbMessages.java

```diff
@@ -0,0 +1,174 @@
+              SmsMessage.newBuilder()
+                  .addAllSuggesters(newSuggesters)
+                  .setId(smsMessage.getId())
+                  .setReceivedAt(smsMessage.getReceivedAt())
+                  .setSender(smsMessage.getSender())
+                  .setBody(smsMessage.getBody())
+                  .setApproved(smsMessage.getApproved())
+                  .build();
+
+          modifyMessage(newMessage);
+        }
+        // Case: User is the only one suggested this spam
+        else {
+          if (!smsMessage.getApproved()) { //TODO make sure the user is admin
+            // if the sms was approved by an admin it will not be deleted. only admin can delete approved spam.
+            messages.remove(smsMessage);
+            client.deleteMessage(smsMessage);
+            notifyListeners(messages.indexOf(smsMessage), smsMessage, Type.REMOVED);
+          } else {
+            Log.e(TAG, "user try to remove approved spam");
+          }
+        }
+      } else {
+        Log.e(TAG, "remove something that the user didn't suggest");
+      }
+    } else {
+      Log.w(
+          TAG,
+          "Attempt to remove message by its ID, but message not found. This might come from inbox trying to find a similiar spam in the db.\nmessage id: "
+              + smsMessage.getId());
+    }
+  }
+
+  public void modifyMessage(SmsMessage newMessage) {
+    int index = findSms(newMessage,messages);
+    if (index != -1) {
+      SmsMessage oldMessage = messages.get(index);
+      messages.set(index, newMessage);
+      client.modifyMessage(oldMessage, newMessage);
+      notifyListeners(index, newMessage, Type.MODIFIED);
+    } else {
+      Log.e(
+          TAG,
+          "Attempt to modify message but message not found.\n Message id: "
+              + newMessage.getId()
+              + "\nBody: "
+              + newMessage.getBody()
+              + "\nSender: "
+              + newMessage.getSender());
+    }
+  }
+
+  public void addMessagesListener(MessagesListener messagesListener) {
+    this.messagesListeners.add(messagesListener);
+  }
+
+  public void notifyListeners(int index, SmsMessage smsMessage, Type type) {
+    for (MessagesListener listener : messagesListeners) {
+      switch (type) {
+        case ADDED:
+          listener.messageAdded(smsMessage);
+          break;
+        case MODIFIED:
+          listener.messageModified(index);
+          break;
+        case REMOVED:
+          listener.messageRemoved(index, smsMessage);
+          break;
+      }
+    }
+  }
+
+  public List<SmsMessage> getMessages() {
+    return messages;
+  }
+
+
+  public interface MessagesListener {
+
+    void messageAdded(SmsMessage newMessage);
+
+    void messageModified(int index);
+
+    // SmsMessage parameter is passed for cases the index already had been removed from the db list
+    void messageRemoved(int index, SmsMessage smsMessage);
+
+  }
+
+  // static function to search a given list
```

Comment:
No need to repeat the code in the comment, so I'd remove `static function`.
I'd rephrase to:
`// Searches in the given list`